example/src/app/component/flat_list.jsx

```diff
@@ -1,25 +1,65 @@
 import React from "react";
 import isEqual from "lodash.isequal";
 
+const defaultBatchCount = 50;
+
 const itemChecker = (prevProps, nextProps) => {
   return isEqual(prevProps, nextProps);
 };
 
-const ItemRenderer = React.memo(({ item, index, renderItem }) => renderItem(item, index), itemChecker);
+const ItemRenderer = React.memo(({ item, renderItem }) => renderItem(item), itemChecker);
 
 class FlatList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: []
+    };
+    this.timeout;
+  }
+
   componentDidMount() {
-    {console.log("After redux-counter",performance.now())}
+    const { data, batchCount = defaultBatchCount, isVirtual = false } = this.props;
+    this.setState({
+      items: isVirtual ? data.slice(0, batchCount) : data
+    });
   }
+
+  componentDidUpdate({ data: oldData }) {
+    const { data, batchCount = defaultBatchCount } = this.props;
+    const { items } = this.state;
+
+    if (oldData.length == data.length) {
+      if (itemChecker({ data: oldData }, { data })) {
+        clearTimeout(this.timeoutId);
+        const hasMore =
+          items.length + batchCount <= data.length ||
+          (data.length - items.length <= batchCount && data.length !== items.length);
+        if (hasMore) {
+          this.timeoutId = setTimeout(() => {
+            this.setState(prevState => ({
+              items: [...items, ...data.slice(prevState.items.length, prevState.items.length + batchCount)]
+            }));
+          }, 100);
```

Comment:
Check if i set 10ms instead of 100 ....will anything be affected ? Make This as a Constant